api/views/__init__.py

```diff
@@ -234,12 +236,26 @@ def terms_accepted_user(request):
         # create new SocialAccount, User, and Profile for the new Relay user from Firefox
         # Since this is a Resource Provider/Server flow and are NOT a Relying Party (RP) of FXA
         # No social token information is stored (no Social Token object created).
-        complete_social_login(request, social_login)
-        # complete_social_login writes ['account_verified_email', 'user_created', '_auth_user_id', '_auth_user_backend', '_auth_user_hash']
-        # on request.session which sets the cookie because complete_social_login does the "login"
-        # The user did not actually log in, logout to clear the session
-        if request.user.is_authenticated:
-            get_account_adapter(request).logout(request)
+        try:
+            complete_social_login(request, social_login)
+            # complete_social_login writes ['account_verified_email', 'user_created', '_auth_user_id', '_auth_user_backend', '_auth_user_hash']
+            # on request.session which sets the cookie because complete_social_login does the "login"
+            # The user did not actually log in, logout to clear the session
+            if request.user.is_authenticated:
+                get_account_adapter(request).logout(request)
+        except NoReverseMatch as e:
+            # TODO: use this logging to fix the underlying issue
+            # https://mozilla-hub.atlassian.net/browse/MPP-3473
+            if "socialaccount_signup" in e.args[0]:
+                logger.error(
+                    str(e),
+                    extra={
+                        "fxa_uid": fxa_uid,
+                        "social_login.state": social_login.state,
+                    },
+                )
```

Comment:
The varied message will make it hard to find this in the logs. The period in `social_login.state` will also cause headaches. I suggest a static string for the message, and the full exception in the extra:

```suggestion
                logger.error(
                    'socialaccount_signup_error'
                    extra={
                        "exception": str(e),
                        "fxa_uid": fxa_uid,
                        "social_login_state": social_login.state,
                    },
                )
```